### PR TITLE
Replace networkID with chainID - Closes #1220

### LINKED ISSUE #1220: Replace networkID with chainID
### Description

Lisk SDK v6.0.0-alpha.3 replaces the `networkID` with `chainID`. Lisk Service internally uses `networkID` internally for multiple purposes, such as network resolution. So, the code base now must be refactored to use the `chainID` instead.

### Acceptance Criteria

- Genesis block when downloaded from the server must now be stored like `./data/${chainID}/genesis_block.json` instead of `./data/${networkIdentifier}/genesis_block.json`
- The response structure for the following endpoints are up-to-date:
  - `/api/v3/network/status`
  - `/api/v3/peers`
  - `/api/status`
- `networkIdentifier` or `networkID` is used nowhere within the code base anymore for the `/v3` API and replaced with `chainID` instead
- `transaction-statistics` uses `chainID` to resolve the network (eg: mainnet, testnet, betanet etc.) [Search for `config.serviceURL`]
- Swagger is up-to-date
- Tests are up-to-date

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1220

### How was it solved?

- [x] Genesis block when downloaded from the server must now be stored like ./data/${chainID}/genesis_block.json instead of ./data/${networkIdentifier}/genesis_block.json
- The response structure for the following endpoints are up-to-date:
- [x] /api/v3/network/status
- [x] /api/v3/peers
- [x] /api/status
- [x] networkIdentifier or networkID is used nowhere within the code base anymore for the /v3 API and replaced with chainID instead
- [x] transaction-statistics uses chainID to resolve the network (eg: mainnet, testnet, betanet etc.) [Search for config.serviceURL]
- [x] Swagger is up-to-date
- [x] Tests are up-to-date

### CODE REVIEW

File: services/blockchain-indexer/methods/dataService/controllers/network.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const getNetworkStatus = async () => {
 	const result = await dataService.getNetworkStatus();
 	return {
 		data: result.data,
-		meta: {},
+		meta: result.meta,
```
Comment (sameersubudhi) [inline services/blockchain-indexer/methods/dataService/controllers/network.js:22]: Please add null checks.

File: services/gateway/apis/http-version3/swagger/definitions/networkStatus.json [hunk 2]
```diff
@@ -92,41 +101,49 @@
...
                                 "fee",
                                 "random",
-                                "legacy"
+                                "legacy",
+                                "interoperability"
                             ]
                         },
                         "description": "Array of all the current registered modules"
                     },
                     "moduleCommands": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/networkStatus.json:110]: The example currently returns `moduleCommands` as a nested array. Please fix it.

File: services/transaction-statistics/config.js [hunk 3]
```diff
@@ -61,10 +61,22 @@ config.queue = {
...
+	},
+	'01000000': {
+		networkName: 'testnet',
+		serviceUrl: 'https://testnet-service.lisk.com',
+
+	},
+	'': {
+		networkName: 'betanet',
+		serviceUrl: 'https://betanet-service.lisk.com',
+	},
```
Comment (sameersubudhi) [inline services/transaction-statistics/config.js:79]: Please refactor as in `services/blockchain-connector/config.js`.

File: services/transaction-statistics/shared/transactionStatistics.js [hunk 4]
```diff
@@ -186,6 +186,10 @@ const getTransactionsStatistics = async params => {
 	const tokens = await db.rawQuery(`SELECT DISTINCT(tokenID) FROM ${txStatisticsIndexSchema.tableName}`);
 	const tokenIDs = tokens.map(e => e.tokenID);
 
+	if (!tokens.length) {
+		return transactionsStatistics;
+	}
```
Comment (sameersubudhi) [inline services/transaction-statistics/shared/transactionStatistics.js:191]: Please remove this.

```
const [{ date: minDate } = {}] = await db.find({ sort: 'date:asc' }, 'date');
const total = minDate ? moment().diff(moment.unix(minDate), params.interval) : 0;
```

File: services/transaction-statistics/shared/utils/metadata.js [hunk 5]
```diff
@@ -15,25 +15,36 @@
...
 const config = require('../../config');
+const { requestConnector } = require('./request');
+
+const logger = Logger();
 
-// TODO: Resolve network from network status response once available
 const getTokenMetadataByID = async (tokenID) => {
-	const network = 'mainnet';
+	const { chainID } = await requestConnector('getNetworkStatus');
+	const network = config.networks[chainID];
```
Comment (sameersubudhi) [inline services/transaction-statistics/shared/utils/metadata.js:29]: Please refactor according to the changes suggested within the config file. Also, please rename the variable `network` to `networkConfig`.

File: services/transaction-statistics/shared/utils/metadata.js [hunk 6]
```diff
@@ -15,25 +15,36 @@
...
-			{ tokenID, network },
+			{ tokenID, chainID, network: networkName },
 		);
 		return tokenMetadata;
 	} catch (error) {
-		const serviceURL = config.serviceURL[network];
+		const { serviceURL } = network;
 		const tokensMetadataURL = `${serviceURL}/api/v3/blockchain/apps/meta/tokens`;
-		const tokenMetadata = await request(tokensMetadataURL, { tokenID, network });
+		const tokenMetadata = await request(tokensMetadataURL, { tokenID, chainID });
```
Comment (sameersubudhi) [inline services/transaction-statistics/shared/utils/metadata.js:47]: Please double-check this change.

File: tests/integration/gateway/status.test.js [hunk 7]
```diff
@@ -16,21 +16,13 @@
 const config = require('../../config');
 const { api } = require('../../helpers/api');
 
-const {
-	statusSchema,
-	readySchema,
-} = require('../../schemas/status.schema');
+const {	readySchema } = require('../../schemas/status.schema');
```
Comment (sameersubudhi) [inline tests/integration/gateway/status.test.js:19]: Kindly revert the changes. We still need the `/api/status` endpoint.

File: tests/schemas/api_v3/networkStatus.schema.js [hunk 8]
```diff
@@ -18,18 +18,17 @@ import regex from './regex';
...
-	id: Joi.string().required(),
-	name: Joi.string().required(),
+const genesisBlockSchema = {
+	fromFile: Joi.string().required(),
 };
 
 const genesisSchema = {
+	block: Joi.object(genesisBlockSchema).required(),
+	bftBatchSize: Joi.number().integer().required(),
 	blockTime: Joi.number().integer().min(0).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/networkStatus.schema.js:28]:
```suggestion
	bftBatchSize: Joi.number().integer().positive().required(),
	blockTime: Joi.number().integer().positive().required(),
```

File: tests/schemas/api_v3/networkStatus.schema.js [hunk 9]
```diff
@@ -18,18 +18,17 @@ import regex from './regex';
...
-	communityIdentifier: Joi.string().required(),
+	chainID: Joi.string().pattern(regex.CHAIN_ID).required(),
 	maxTransactionsSize: Joi.number().integer().min(0).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/networkStatus.schema.js:30]:
```suggestion
	maxTransactionsSize: Joi.number().integer().positive().required(),
```

File: tests/schemas/api_v3/networkStatus.schema.js [hunk 10]
```diff
@@ -38,15 +37,15 @@ const seedPeerSchema = {
...
 };
 
 const networkStatusSchema = {
 	version: Joi.string().pattern(regex.SEMVER).required(),
 	networkVersion: Joi.string().required(),
-	networkIdentifier: Joi.string().min(1).max(64).pattern(regex.HASH_SHA256)
-		.required(),
+	chainID: Joi.string().pattern(regex.CHAIN_ID).required(),
 	lastBlockID: Joi.string().min(1).max(64).pattern(regex.HASH_SHA256)
 		.required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/networkStatus.schema.js:50]: Also, please update the regex to be exactly 64 characters: `const HASH_SHA256 = /^\b([A-Fa-f0-9]){64}\b$/;`
```suggestion
	lastBlockID: Joi.string().pattern(regex.HASH_SHA256).required(),
```

File: tests/schemas/api_v3/networkStatus.schema.js [hunk 11]
```diff
@@ -55,7 +54,7 @@ const networkStatusSchema = {
 	unconfirmedTransactions: Joi.number().integer().min(0).required(),
 	genesis: Joi.object(genesisSchema).required(),
 	registeredModules: Joi.array().items(Joi.string()).required(),
-	moduleCommands: Joi.array().items(moduleCommandSchema).required(),
+	moduleCommands: Joi.array().items(Joi.string()).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/networkStatus.schema.js:57]: Also, kindly remove `MODULE_COMMAND_ID` and `MODULE_COMMAND_NAME`. Ideally, `MODULE_COMMAND_ID` shouldn't be getting used elsewhere. `MODULE_COMMAND_NAME` can be replaced with `MODULE_COMMAND`.
```suggestion
	moduleCommands: Joi.array().items(Joi.string().pattern(regex.MODULE_COMMAND)).required(),
```

File: tests/schemas/api_v3/peer.schema.js [hunk 12]
```diff
@@ -31,7 +32,7 @@ const peerSchema = {
 	networkVersion: Joi.string().required(),
 	state: Joi.string().valid(...allowedPeerStateNames).required(),
 	height: Joi.number().optional(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/peer.schema.js:34]:
```suggestion
	height: Joi.number().positive().optional(),
```

File: tests/schemas/status.schema.js [hunk 13]
```diff
@@ -15,15 +15,6 @@
  */
 import Joi from 'joi';
 
-const statusSchema = {
```
Comment (sameersubudhi) [inline tests/schemas/status.schema.js:18]: Please revert this change as we still need this.

File: tests/schemas/api_v3/transactionStatistics.schema.js [hunk 14]
```diff
@@ -40,10 +40,10 @@ const transactionStatisticsSchema = {
...
 };
 
-const date = {
-	dateFormat: Joi.string().valid(...allowedDateFormats).required(),
-	dateFrom: Joi.string().required(),
-	dateTo: Joi.string().required(),
+const duration = {
+	format: Joi.string().valid(...allowedDateFormats).required(),
+	from: Joi.string().required(),
+	to: Joi.string().required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionStatistics.schema.js:46]: Please add a new regex: `DURATION: /^\d{4}-\d{2}(?:-\d{2})?$/`

```suggestion
	from: Joi.string().pattern(regex.DURATION).required(),
	to: Joi.string().pattern(regex.DURATION).required(),
```

File: services/blockchain-connector/package.json [hunk 15]
```diff
@@ -35,6 +35,8 @@
     "@liskhq/lisk-cryptography": "4.0.0-alpha.3",
     "big-json": "^3.1.0",
     "bluebird": "^3.7.2",
+    "decompress": "^4.2.1",
+    "decompress-targz": "^4.1.1",
```
Comment (sameersubudhi) [inline services/blockchain-connector/package.json:39]: Please try to get `tar` dependency working else let's figure out something else. These dependencies have not been maintained for a very long time.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-connector/config.js [hunk 16]
```diff
@@ -42,23 +42,26 @@ config.liskAppDataPath = process.env.LISK_APP_DATA_PATH || '~/.lisk/lisk-core';
   * Network-related settings
   */
 config.genesisBlockUrl = process.env.GENESIS_BLOCK_URL || '';
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply services/blockchain-connector/config.js]: applied.

File: services/blockchain-connector/shared/sdk/blocksUtils.js [hunk 17]
```diff
@@ -42,28 +42,27 @@ const getGenesisBlockId = () => genesisBlock.header.id;
 
 const loadConfig = async () => {
 	const nodeInfo = await getNodeInfo();
-	const { networkIdentifier } = nodeInfo;
-
-	if (process.env.GENESIS_BLOCK_URL) {
-		logger.info('Genesis block URL is defined by environment variable (GENESIS_BLOCK_URL)');
+	const { chainID } = nodeInfo;
 
+	if (config.genesisBlockUrl !== '') {
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/blocksUtils.js]: applied.

File: services/blockchain-indexer/methods/dataService/controllers/network.js (hunk 1 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/methods/dataService/controllers/network.js]: applied.

File: services/gateway/apis/http-version3/swagger/definitions/networkStatus.json (hunk 2 above)
Comment (priojeetpriyom) [reply services/gateway/apis/http-version3/swagger/definitions/networkStatus.json:110]: applied.

File: services/transaction-statistics/config.js (hunk 3 above)
Comment (priojeetpriyom) [reply services/transaction-statistics/config.js]: applied

File: services/transaction-statistics/shared/transactionStatistics.js (hunk 4 above)
Comment (priojeetpriyom) [reply services/transaction-statistics/shared/transactionStatistics.js]: applied.

File: services/transaction-statistics/shared/utils/metadata.js (hunk 5 above)
Comment (priojeetpriyom) [reply services/transaction-statistics/shared/utils/metadata.js]: applied.

File: tests/schemas/api_v3/networkStatus.schema.js (hunk 8 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/networkStatus.schema.js]: applied.

File: tests/schemas/api_v3/networkStatus.schema.js (hunk 10 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/networkStatus.schema.js]: applied.

File: tests/schemas/api_v3/peer.schema.js (hunk 12 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/peer.schema.js]: applied.

File: tests/schemas/api_v3/transactionStatistics.schema.js (hunk 14 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionStatistics.schema.js]: applied.

File: services/blockchain-app-registry/config.js [hunk 18]
```diff
@@ -65,10 +65,11 @@ config.gitHub = {
...
 config.dataDir = `${process.cwd()}/data`;
 
+// TODO: update betanet chainID once determined
 config.serviceURL = {
-	mainnet: process.env.SERVICE_URL_MAINNET || 'https://service.lisk.com',
-	testnet: process.env.SERVICE_URL_TESTNET || 'https://testnet-service.lisk.com',
-	betanet: process.env.SERVICE_URL_BETANET || 'https://betanet-service.lisk.com',
+	'00000000': process.env.SERVICE_URL_MAINNET || 'https://service.lisk.com',
+	'01000000': process.env.SERVICE_URL_TESTNET || 'https://testnet-service.lisk.com',
+	'': process.env.SERVICE_URL_BETANET || 'https://betanet-service.lisk.com',
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/config.js:72]: Please revert this change as well as previously discussed.

File: tests/schemas/api_v3/transactionStatistics.schema.js [hunk 19]
```diff
@@ -40,10 +40,10 @@ const transactionStatisticsSchema = {
 	distributionByAmount: Joi.object().required(),
 };
 
-const date = {
-	dateFormat: Joi.string().valid(...allowedDateFormats).required(),
-	dateFrom: Joi.string().required(),
-	dateTo: Joi.string().required(),
+const duration = {
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionStatistics.schema.js:43]:
```suggestion
const durationSchema = {
```

File: tests/schemas/api_v3/transactionStatistics.schema.js [hunk 20]
```diff
@@ -64,7 +64,7 @@ const metaSchema = {
 	limit: Joi.number().required(),
 	offset: Joi.number().required(),
 	total: Joi.number().required(),
-	date: Joi.object(date).required(),
+	duration: Joi.object(duration).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionStatistics.schema.js:67]:
```suggestion
	duration: Joi.object(durationSchema).required(),
```

File: services/blockchain-app-registry/config.js (hunk 18 above)
Comment (priojeetpriyom) [reply services/blockchain-app-registry/config.js]: the `serviceURL` is not being used anymore. removing all together.

File: tests/schemas/status.schema.js (hunk 13 above)
Comment (priojeetpriyom) [reply tests/schemas/status.schema.js]: applied.

File: tests/schemas/api_v3/transactionStatistics.schema.js (hunk 19 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionStatistics.schema.js]: applied

File: tests/schemas/api_v3/transactionStatistics.schema.js (hunk 20 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionStatistics.schema.js]: applied.

File: services/blockchain-app-registry/config.js (hunk 18 above)
Comment (priojeetpriyom) [reply services/blockchain-app-registry/config.js]: applied.

File: tests/schemas/api_v3/tokens.schema.js [hunk 21]
```diff
@@ -17,7 +17,7 @@ import Joi from 'joi';
 import regex from './regex';
 
 const lockedAmount = {
-	moduleID: Joi.string().pattern(regex.MODULE_COMMAND_ID).required(),
+	module: Joi.string().pattern(regex.MODULE_COMMAND).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/tokens.schema.js:20]: `module` would not satisfy the pattern.
```suggestion
	module: Joi.string().required(),
```

Comment (priojeetpriyom) [reply tests/schemas/api_v3/tokens.schema.js:20]: use `regex.MODULE` which is similar to `regex.NAME`

File: tests/schemas/api_v3/networkStatus.schema.js (hunk 11 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/networkStatus.schema.js:57]: Applied.

File: tests/schemas/api_v3/tokens.schema.js (hunk 21 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/tokens.schema.js]: Applied.

File: tests/schemas/api_v3/networkStatus.schema.js (hunk 11 above)
Comment (sameersubudhi) [reply tests/schemas/api_v3/networkStatus.schema.js:57]: The pattern check is still missing. Please update.
```suggestion
	moduleCommands: Joi.array().items(Joi.string().pattern(regex.MODULE_COMMAND)).required(),
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: tests/schemas/api_v3/regex.js [hunk 22]
```diff
@@ -27,15 +25,18 @@ const INTERVAL = /^\b((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))(:
 const FILE_NAME = /^\btransactions_(lsk[a-hjkm-z2-9]{38})_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))\.csv\b$/;
 const FILE_URL = /^\/api\/v2\/exports\/transactions_(lsk[a-hjkm-z2-9]{38})_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))_((\d{4})-((1[012])|(0?[1-9]))-(([012][1-9])|([123]0)|31))\.csv$/;
 const NETWORK = /^\b(?:mainnet|testnet|betanet){1}\b$/;
+const MODULE = /^[\w!@$&.]{1,20}$/;
 const MODULE_COMMAND = /[a-zA-Z]{1,32}:[a-zA-Z]{1,32}/;
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/schemas/api_v3/regex.js]: Applied.

File: tests/schemas/api_v3/networkStatus.schema.js (hunk 11 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/networkStatus.schema.js]: Applied.

File: services/blockchain-connector/package.json (hunk 15 above)
Comment (priojeetpriyom) [reply services/blockchain-connector/package.json]: resolved with `tar`. Thanks a lot @sameersubudhi for the 🪄

File: services/blockchain-connector/shared/utils/fs.js [hunk 23]
```diff
@@ -62,9 +62,10 @@ const rm = async (directoryPath, options = {}) => {
 	);
 };
 
-const extractTarBall = async (filePath, directoryPath) => fs
-	.createReadStream(filePath)
-	.pipe(tar.extract({ cwd: directoryPath }));
+const extractTarBall = async (filePath, directoryPath) => decompress(filePath, directoryPath)
+	.then(() => {
+		logger.debug(`Successfully compressed file: ${directoryPath}`);
+	});
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/utils/fs.js]: Applied.

File: services/blockchain-connector/config.js [hunk 24]
```diff
@@ -41,24 +42,29 @@ config.liskAppDataPath = process.env.LISK_APP_DATA_PATH || '~/.lisk/lisk-core';
...
-	},
-	{
-		name: 'betanet',
-		identifier: '15f0dacc1060e91818224a94286b13aa04279c640bd5d6f193182031d133df7c',
-		genesisBlockUrl: 'https://downloads.lisk.com/lisk/betanet/genesis_block.json.tar.gz',
-	},
-];
+config.constants.GENESIS_BLOCK_URL_DEFAULT = '';
+config.genesisBlockUrl = process.env.GENESIS_BLOCK_URL
+							|| config.constants.GENESIS_BLOCK_URL_DEFAULT;
```
Comment (nagdahimanshu) [inline services/blockchain-connector/config.js:47]: Please fix the alignment.
```suggestion
config.constants.GENESIS_BLOCK_URL_DEFAULT = '';
config.genesisBlockUrl = process.env.GENESIS_BLOCK_URL || config.constants.GENESIS_BLOCK_URL_DEFAULT;
```

Comment (priojeetpriyom) [reply services/blockchain-connector/config.js:47]: resolved in offline discussion.

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Review by sameersubudhi — APPROVED** [review]
